### ISSUE TITLE
[css-fonts] Should font-variant-emoji affect ZWJ emoji sequences?

### ISSUE THREAD

**Issue by tursunova** [w3c/csswg-drafts#11036]
There are some ZWJ emoji sequences that do not have VS, compare https://www.unicode.org/Public/emoji/latest/emoji-zwj-sequences.txt. Should `font-variant-emoji` property affect any of the codepoints in these sequences?

Let's consider the following use case:

```<div style="font-variant-emoji: text">&#x1F408;&#x200D;&#x2B1B;</div>```

According to the [CSS Fonts Module 4 spec](https://www.w3.org/TR/css-fonts-4/#font-variant-emoji-prop):

>Only the code points [listed](http://www.unicode.org/emoji/charts/emoji-variants.html) by Unicode as contributing to a Unicode emoji presentation sequence are affected by this property.

U+1F408 is listed there, since U+1F408 U+FE0E is a valid variation sequence, compare: https://www.unicode.org/Public/15.1.0/ucd/emoji/emoji-variation-sequences.txt.

So the way I interpret spec right now, it example above should act as:

```<div">&#x1F408;&#xFE0E;&#x200D;&#x2B1B;</div>```

and split the zwj sequence.

Is my understanding correct or do we want `font-variant-emoji` to not affect any of the codepoints in zwj sequences? 

@drott @svgeesus

**Comment by svgeesus** [comment]
@xfq any guidance?

**Comment by yisibl** [comment]
@markusicu We need your professional guidance too.

**Comment by jfkthame** [comment]
> There are some ZWJ emoji sequences that do not have VS, compare https://www.unicode.org/Public/emoji/latest/emoji-zwj-sequences.txt. Should `font-variant-emoji` property affect any of the codepoints in these sequences?

Yes, in my opinion: the natural expectation would be that any character or sequence that displays an emoji-style glyph by default should change to a text-style glyph if `font-variant-emoji: text` is applied.

> Let's consider the following use case:
> 
> `<div style="font-variant-emoji: text">&#x1F408;&#x200D;&#x2B1B;</div>`
> 
> According to the [CSS Fonts Module 4 spec](https://www.w3.org/TR/css-fonts-4/#font-variant-emoji-prop):
> 
> > Only the code points [listed](http://www.unicode.org/emoji/charts/emoji-variants.html) by Unicode as contributing to a Unicode emoji presentation sequence are affected by this property.
> 
> U+1F408 is listed there, since U+1F408 U+FE0E is a valid variation sequence, compare: https://www.unicode.org/Public/15.1.0/ucd/emoji/emoji-variation-sequences.txt.
> 
> So the way I interpret spec right now, it example above should act as:
> 
> `<div">&#x1F408;&#xFE0E;&#x200D;&#x2B1B;</div>`
> 
> and split the zwj sequence.

The spec wording is perhaps unfortunate here, because I think users will assume that `font-variant-emoji` applies to "emoji text units", which may be individual codepoints *or* sequences; that distinction is not useful to users.

So in the case of `&#x1F408;&#x200D;&#x2B1B;`, applying `font-variant-emoji: text` does not insert an actual `U+FE0E` codepoint and break the sequence (resulting in a text-style cat, perhaps as a line-drawing or outline, followed by a black square); it's still an emoji sequence requesting a black cat, but now it's supposed to be a text-style black cat rather than an emoji-style one.

**Comment by drott** [comment]
> Yes, in my opinion: the natural expectation would be that any character or sequence that displays an emoji-style glyph by default should change to a text-style glyph if font-variant-emoji: text is applied.

The issue is how would such sequences be encoded in a font? Inserting VS15 would make the sequences non RGI sequences. We're currently looking into sequences support in our Noto Emoji (non-color) font. Are there any other known monochromatic fonts that would have support for sequences?

> So in the case of &#x1F408;&#x200D;&#x2B1B;, applying font-variant-emoji: text does not insert an actual U+FE0E codepoint and break the sequence (resulting in a text-style cat, perhaps as a line-drawing or outline, followed by a black square); it's still an emoji sequence requesting a black cat, but now it's supposed to be a text-style black cat rather than an emoji-style one.

The way I read the spec:
https://www.w3.org/TR/css-fonts-4/#valdef-font-variant-emoji-text
for **text**: "Code points are rendered as if U+FE0E VARIATION SELECTOR-15 was appended to every Presentation Participating Code Point."
this means it would virtually insert VS15 after cat and break the sequence into a non RGI sequence.

In any case, we are back at what is the encoding the font and how do we differentiate a color from a non-color encoding for ZWJ sequences.

**Comment by drott** [comment]
This [sample of Noto Emoji](https://fonts.google.com/noto/specimen/Noto+Emoji?preview.text=%F0%9F%90%88%E2%80%8D%E2%AC%9B%F0%9F%90%88%F0%9F%91%A9%E2%80%8D%F0%9F%91%A9%E2%80%8D%F0%9F%91%A6%E2%80%8D%F0%9F%91%A6) shows that black cat is distinguished from white cat in that font. On a cursory test, I did not find RGI ZWJ sequences supported that contain VS16.

**Comment by Crissov** [comment]
Isn‘t the whole point of `font-variant-emoji` that it can switch the default presentation style of emojis *as perceived by readers* without having to deal with VSs which can by their very nature only apply to a single character (possibly composed of a base and combining ones), but *not* a sequence?!

That means, ZWJ sequences, RIS (flags), skin tone modifiers and enclosing sequences (keycaps) all must be able to be affected by this property, unless they actually contain VS15 or VS16.
